rewrite-core/src/main/java/org/openrewrite/SourceFile.java

```diff
@@ -79,6 +82,21 @@ default <S extends Style> S getStyle(Class<S> style, S defaultStyle) {
         return s == null ? defaultStyle : s;
     }
 
+    default <S extends Style> S getStyleOrDefault(Class<S> style, Supplier<S> defaultStyle) {
```

Comment:
I JavaSourceFile we have overridden service() to support an AutoFormatService. Possibly we should pull that up here. That would then not require adding yet more methods to this API.